src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ProfileDataNode.cs

```diff
@@ -125,5 +125,17 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
         {
             return Array.Empty<DependencyListEntry>();
         }
+
+        public override int CompareToImpl(ISortableNode other, CompilerComparer comparer)
+        {
+            ProfileDataNode otherNode = (ProfileDataNode)other;
+            int result = _ilSize.CompareTo(otherNode._ilSize);
```

Comment:
There's only one ProfileDataNode per MethodWithGCInfo. It's enough to compare the _methodNode.